pandas/tests/reshape/concat/test_concat.py

```diff
@@ -858,3 +858,17 @@ def test_concat_multiindex_with_category():
     )
     expected = expected.set_index(["c1", "c2"])
     tm.assert_frame_equal(result, expected)
+
+
+def test_concat_datetime64_diff_resolution():
```

Comment:
might be simpler or more thorough to take existing tests in this file and parametrize them over units?